### ISSUE TITLE
UtamBase.waitFor overload to allow specifying timeout explicitly

### ISSUE THREAD

**Issue by alan-morey** [salesforce/utam-java#167]
### Is your feature request related to a problem? Please describe

`UtamBase.waitFor` does not provide options to explicitly set a timeout that differs from the globally configured timeout I set with `UtamLoaderConfig.setExplicitTimeout`

I would like `UtamBase.waitFor` to be overloaded with same signature as `Driver.waitFor`, so that I can specify a wait condition with a custom timeout for that wait operation.

Currently I'm increasing my global explicit timeout with `UtamLoaderConfig.setExplicitTimeout` but this is increasing the overall test execution time when there is an error/failure in the test and the wait condition will never be true.

In some cases, I may want to wait a little longer (or shorter) than the global timeout. If an action is purely client side then a shorter wait could be appropriate, if action results in some external web service call and I need to wait for the results, then a longer wait time would be more appropriate.

### Describe the solution you'd like

I would like `UtamBase` interface (and implementations) to include a `waitFor` overload with additional options, similar to `Driver.waitFor`

e.g.
```
public interface UtamBase {
  ...

  /**
   * polling wait repeatedly applies expectations until truthy value is return (not null or boolean
   * true)
   *
   * @param timeout timeout after which exception is thrown if condition is not met. If passed as
   *                null, timeout from config is used
   * @param isTrue  condition to apply
   * @param message error message to throw if timeout is reached, can be null
   * @param <T>     return type
   * @return result of the applied expectations
   */
  <T> T waitFor(Supplier<T> isTrue, String message, Duration timeout);

  ...
}
```

### Describe alternatives you've considered

Maybe use Selenium `FluentWait` to achieve the same but then I'm jumping out of the UTAM abstractions. Alternatively I could use the UTAM `Driver` but seems like there is no way to access that from a `UtamBase` or the `UtamLoader`.

On a side note, should `UtamBase` and `UtamLoader` have a public `getDriver()` method?

I'm currently setting up `UtamLoader` as follows:
```
final UtamLoader setupUtamLoader(WebDriver driver) {
    var config = new UtamLoaderConfigImpl("utam-loader.config.json");
    config.setExplicitTimeout(Duration.ofSeconds(90));
    config.setImplicitTimeout(Duration.ZERO);

    utamLoader = new UtamLoaderImpl(config, driver);

    return utamLoader;
}
```

**Comment by jimevanssfdc** [comment]
@alan-morey This seems like a reasonable feature request, and the UTAM team will take it under advisement for implementation in a future release. In the meantime, you can use your existing `UtamLoader` instance to set the length of the explicit wait timeout at runtime. Something like the following (N.B., code written from memory, and without the benefit of an IDE, so may not work exactly correctly, or even compile if copy-pasted directly into your project):

public class MyTests {

    private UtamLoader loader;

    @BeforeAll
    public void setup() {
        // Call your loader setup method, with a valid WebDriver instance.
        // Creation of WebDriver instance is omitted for brevity.
        loader = setupUtamLoader(driver);
    }

    @Test
    public void myTest() {
        // Load the root PO, which we will assume has the waitFor you'd like to configure
        MyPageObject pageObjectInstance = loader.load(MyPageObject.class);

        // Dynamically set the explicit timeout of the loader config
        loader.getConfig().setExplicitTimeout(Duration.ofSeconds(5));
        loader.resetContext();

        // Call the waitFor method on the appropriate element
        pageObjectInstance.getMyElement().waitFor(...);

        // Reset the loader config to the larger timeout setting
        loader.getConfig().setExplicitTimeout(Duration.ofSeconds(90));
        loader.resetContext();
    }
}

**Comment by irinash13** [comment]
@alan-morey - what company do you represent?

**Comment by lizaiv77** [comment]
@alan-morey Sorry for late response, feature was implemented in version 1.6.0
Docs https://utam.dev/grammar/explicit-waits
